kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/TypeRenderer.kt

```diff
@@ -66,12 +66,21 @@ abstract class TypeRenderer {
               Types::class,
               render(typeName.typeArguments[0].objectType()))
         } else {
-          val placeholders = typeName.typeArguments.joinToString(", ") { "%L" }
-          CodeBlock.of(
-              "%T.newParameterizedType(%T::class.java, $placeholders)",
-              Types::class,
-              typeName.rawType.objectType(),
-              *(typeName.typeArguments.map { render(it.objectType()) }.toTypedArray()))
+          val builder = CodeBlock.builder().apply {
```

Comment:
We should add buildCodeBlock to KotlinPoet to match buildString and buildSpannedString

Reply:
https://github.com/square/kotlinpoet/issues/468